solidity/test/bank/Bank.test.ts

```diff
@@ -0,0 +1,1109 @@
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should transfer the requested amount", async () => {
+        expect(await bank.balanceOf(spender.address)).to.equal(
+          initialBalance.sub(amount)
+        )
+        expect(await bank.balanceOf(recipient)).to.equal(amount)
+      })
+
+      it("should emit the BalanceTransferred event", async () => {
+        await expect(tx)
+          .to.emit(bank, "BalanceTransferred")
+          .withArgs(spender.address, recipient, amount)
+      })
+    })
+
+    context(
+      "when the spender transfers part of their balance in two transactions",
+      () => {
+        const amount1 = to1e18(21)
+        const amount2 = to1e18(12)
+
+        before(async () => {
+          await createSnapshot()
+          await bank.connect(spender).transferBalance(recipient, amount1)
+          await bank.connect(spender).transferBalance(recipient, amount2)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should transfer the requested amount", async () => {
+          expect(await bank.balanceOf(spender.address)).to.equal(
+            initialBalance.sub(amount1).sub(amount2)
+          )
+          expect(await bank.balanceOf(recipient)).to.equal(amount1.add(amount2))
+        })
+      }
+    )
+
+    context("when the spender transfers their entire balance", () => {
+      const amount = initialBalance
+      let tx: ContractTransaction
+
+      before(async () => {
+        await createSnapshot()
+        tx = await bank.connect(spender).transferBalance(recipient, amount)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should transfer the entire balance", async () => {
+        expect(await bank.balanceOf(spender.address)).to.equal(0)
+        expect(await bank.balanceOf(recipient)).to.equal(amount)
+      })
+
+      it("should emit the BalanceTransferred event", async () => {
+        await expect(tx)
+          .to.emit(bank, "BalanceTransferred")
+          .withArgs(spender.address, recipient, amount)
+      })
+    })
+
+    context("when the spender transfers 0 balance", () => {
+      const amount = 0
+      let tx: ContractTransaction
+
+      before(async () => {
+        await createSnapshot()
+        tx = await bank.connect(spender).transferBalance(recipient, amount)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should transfer no balance", async () => {
+        expect(await bank.balanceOf(spender.address)).to.equal(initialBalance)
+        expect(await bank.balanceOf(recipient)).to.equal(0)
+      })
+
+      it("should emit the BalanceTransferred event", async () => {
+        await expect(tx)
+          .to.emit(bank, "BalanceTransferred")
+          .withArgs(spender.address, recipient, 0)
+      })
+    })
+  })
+
+  describe("approve", () => {
```

Comment:
It should be named "approveBalance".

Reply:
😶 https://github.com/keep-network/tbtc-v2/pull/84/commits/2a97d82569ac798d9b423f10b3023a6744ac4283